### PR TITLE
Android login

### PR DESCRIPTION
**Pull request by lar-ryan-nick**
Created first draft of login page for android

### CODE REVIEW

File: android/app/src/main/java/com/tourneynizer/tourneynizer/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.tourneynizer.tourneynizer;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.text.Editable;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+
+public class MainActivity extends AppCompatActivity {
```
Comment (ankushrayabhari) [inline android/app/src/main/java/com/tourneynizer/tourneynizer/MainActivity.java]: Is there a reason that you chose to extend AppCompatActivity here?

File: android/app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		android:text="Log In"
+		android:textAlignment="center"
+		android:layout_marginTop="30dp"
+		android:textSize="40sp"/>
+
+	<TextView
+		android:layout_width="match_parent"
+		android:layout_height="wrap_content"
+		android:layout_marginHorizontal="10dp"
+		android:text="Email:"
```
Comment (ankushrayabhari) [inline android/app/src/main/res/layout/activity_main.xml]: Reference strings.xml here instead of hardcoding inside the xml file.

File: android/app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		android:layout_marginHorizontal="10dp"
+		android:textSize="20sp"
+		android:id="@+id/username"
+		/>
+
+	<TextView
+		android:layout_width="match_parent"
+		android:layout_height="wrap_content"
+		android:layout_marginHorizontal="10dp"
+		android:text="Password:"
```
Comment (ankushrayabhari) [inline android/app/src/main/res/layout/activity_main.xml]: Reference strings.xml here instead of hardcoding inside the xml file.

File: android/app/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+		android:inputType="textPassword"
+		android:layout_marginHorizontal="10dp"
+		android:textSize="20sp"
+		android:id="@+id/password"
+		/>
+
+	<Button
+		android:layout_width="wrap_content"
+		android:layout_height="wrap_content"
+		android:text="GO"
```
Comment (ankushrayabhari) [inline android/app/src/main/res/layout/activity_main.xml]: Reference strings.xml here instead of hardcoding inside the xml file. Change GO to login.

File: android/app/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	<EditText
+		android:layout_width="match_parent"
+		android:layout_height="wrap_content"
+		android:inputType="textPassword"
+		android:layout_marginHorizontal="10dp"
+		android:textSize="20sp"
+		android:id="@+id/password"
+		/>
+
+	<Button
```
Comment (ankushrayabhari) [inline android/app/src/main/res/layout/activity_main.xml]: Also include a register button.

File: android/app/src/main/java/com/tourneynizer/tourneynizer/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+public class MainActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        Button loginButton = findViewById(R.id.loginButton);
+        loginButton.setOnClickListener(new View.OnClickListener() {
+            @Override
```
Comment (pmh192) [inline android/app/src/main/java/com/tourneynizer/tourneynizer/MainActivity.java]: Can we use lambda functions here?

Comment (ankushrayabhari) [reply android/app/src/main/java/com/tourneynizer/tourneynizer/MainActivity.java]: Don't think we can under current API level.

File: android/app/src/main/res/layout/activity_main.xml (hunk 5 above)
Comment (lar-ryan-nick) [reply android/app/src/main/res/layout/activity_main.xml]: I am planning on making a separate register activity since it requires the name field as well. I will add the register button on the login activity, but it will redirect the user to the "create account" activity. I will also make the create account activity the default activity after I create it.

Comment (ankushrayabhari) [reply android/app/src/main/res/layout/activity_main.xml]: I'd say that login should be the default activity if not signed in because people could be coming from the website/android.

File: android/app/src/main/res/layout/activity_main.xml (hunk 4 above)
Comment (lar-ryan-nick) [reply android/app/src/main/res/layout/activity_main.xml]: But then both the page title and button will have the same text. Are you ok with that?

File: android/app/src/main/java/com/tourneynizer/tourneynizer/MainActivity.java (hunk 1 above)
Comment (lar-ryan-nick) [reply android/app/src/main/java/com/tourneynizer/tourneynizer/MainActivity.java]: This was auto generated. I believe that it has something to do with the action bar at the top of the UI. The android API said that with AppCompatActivity, I can add functionality to the action bar. For now I'll just use a regular Activity, but may return to it later for added functionality.

Comment (lar-ryan-nick) [reply android/app/src/main/java/com/tourneynizer/tourneynizer/MainActivity.java]: Turns out that without AppCompatActivity, the action bar dissapears so I am going to add it back.

File: android/app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -1,3 +1,8 @@
 <resources>
     <string name="app_name">Tourneynizer</string>
+    <string name="log_in">Log In</string>
```
Comment (ankushrayabhari) [inline android/app/src/main/res/values/strings.xml]: This is one word. "Login"

File: android/app/src/main/res/layout/activity_main.xml [hunk 8]
```diff
@@ -7,48 +7,66 @@
...
+			android:layout_width="wrap_content"
+			android:layout_height="wrap_content"
+			android:text="@string/log_in"
+            android:layout_centerInParent="true"
+			android:id="@+id/loginButton"
+		/>
+
+	</RelativeLayout>
+
+	<TextView
```
Comment (ankushrayabhari) [inline android/app/src/main/res/layout/activity_main.xml]: I was thinking maybe just a simple button that says "Register" or "Create Account".

Comment (lar-ryan-nick) [reply android/app/src/main/res/layout/activity_main.xml]: Like the Roastr iOS setup?

Comment (ankushrayabhari) [reply android/app/src/main/res/layout/activity_main.xml]: Not sure. But it seems too wordy right now and could be reduced. The button would be a more apt view then.
